install.sh

```diff
@@ -262,6 +262,59 @@ outdated_glibc() {
   version_lt "$glibc_version" "$REQUIRED_GLIBC_VERSION"
 }
 
+usage() {
+  cat <<EOS
+Homebrew Installer
+Usage: /bin/bash install.sh [--help] [--brew-git-remote URL] [--core-git-remote URL]
+
+This script installs Homebrew, the missing package manager for macOS (or Linux).
+
+Options:
+  -b, --brew-git-remote=URL   Use this URL as the Homebrew/brew git remote.
+                              Default: ${HOMEBREW_BREW_DEFAULT_GIT_REMOTE}.
+
+  -c, --core-git-remote=URL   Use this URL as the Homebrew/homebrew-core git remote.
+                              Default: ${HOMEBREW_CORE_DEFAULT_GIT_REMOTE}.
+
+  -h, --help                  Show this help message and exit.
+EOS
+  exit "${1:-0}"
+}
+
+# Parse remote URLs of Homebrew repositories from command line arguments if presented.
+unset HOMEBREW_{BREW,CORE}_GIT_REMOTE  # unset variables from environment at first
+# The value of the script basename in different situations:
+#   - "bash":       invoked by `bash -c` with no arguments.
+#   - "install.sh": invoked by bash directly.
+#   - otherwise:    invoked by `bash -c` with at least one argument.
+if [[ "$(basename -- "$0")" != "bash" && "$(basename -- "$0")" != "install.sh" ]]; then
+  # The script is invoked by `bash -c "$(cat install.sh)" ...` with at least one argument.
+  # Unshift the first argument back to the argument list
+  set -- "$0" "$@"
+fi
+while [[ "$#" -gt 0 ]]; do
+  opt="$1"
+  shift  # expose the next argument
+  case "$opt" in
+    --) break ;;
```

Comment:
I think we don't need this `--` tbh because this script never passes control to other scripts or programs.